src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+    private static final int FIRST_ARRAY_ELEMENT = 0;
+    private static final int SECOND_ARRAY_ELEMENT = 1;
+    private static final int THIRD_ARRAY_ELEMENT = 2;
+    private static final int FOURTH_ARRAY_ELEMENT = 3;
+    private static final int ONE_DAY = 1;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder sb = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+        String[] splitDateFrom = dateFrom.split("\\.");
+        String[] splitDateTo = dateTo.split("\\.");
+        LocalDate dateBefore = LocalDate.of(Integer.parseInt(splitDateFrom[THIRD_ARRAY_ELEMENT]),
+                Integer.parseInt(splitDateFrom[SECOND_ARRAY_ELEMENT]),
+                Integer.parseInt(splitDateFrom[FIRST_ARRAY_ELEMENT]));
+        LocalDate dateAfter = LocalDate.of(Integer.parseInt(splitDateTo[THIRD_ARRAY_ELEMENT]),
+                Integer.parseInt(splitDateTo[SECOND_ARRAY_ELEMENT]),
+                Integer.parseInt(splitDateTo[FIRST_ARRAY_ELEMENT]));
+
+        for (String name:names) {
+            int sumSalary = 0;
+            for (String dataLine:data) {
+                String[] dataLineParts = dataLine.split(" ");
+                String[] dataLinePartWithDate = dataLineParts[FIRST_ARRAY_ELEMENT].split("\\.");
+                LocalDate workDate = LocalDate.of(Integer
+                        .parseInt(dataLinePartWithDate[THIRD_ARRAY_ELEMENT]),
+                        Integer.parseInt(dataLinePartWithDate[SECOND_ARRAY_ELEMENT]),
+                        Integer.parseInt(dataLinePartWithDate[FIRST_ARRAY_ELEMENT]));
```

Comment:
Those constant names don't give us more information than just indexes. We use constants to add context to the code. So the right name will look like this: USER_NAME_INDEX, DATE_INDEX, etc